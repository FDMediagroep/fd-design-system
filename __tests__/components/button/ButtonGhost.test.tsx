import React from 'react';
import {
    ButtonGhost,
    getCssClassNames,
} from '../../../src/components/button/ButtonGhost';
import { render, fireEvent } from '@testing-library/react';

describe('ButtonGhost', () => {
    test('should render correctly', () => {
        const { getByText } = render(<ButtonGhost>Knopje</ButtonGhost>);

        expect(getByText('Knopje')).toBeTruthy();
    });

    test('should handle clicks correctly', () => {
        const spy = jest.fn();
        const { getByText, container } = render(
            <ButtonGhost onClick={spy}>Knopje</ButtonGhost>
        );

        expect(getByText('Knopje')).toBeTruthy();
        expect(spy).toHaveBeenCalledTimes(0);
        fireEvent.click(container.querySelector('button')!);
        expect(spy).toHaveBeenCalledTimes(1);
        fireEvent.click(container.querySelector('button')!);
        expect(spy).toHaveBeenCalledTimes(2);
    });

    test('should handle sizes correctly', () => {
        const { container } = render(
            <ButtonGhost size="s">Knopje</ButtonGhost>
        );
        const { container: container2 } = render(
            <ButtonGhost size="m">Knopje</ButtonGhost>
        );

        expect(container).toMatchSnapshot();
        expect(container2).toMatchSnapshot();
    });

    test('should have correct class name', () => {
        const { container } = render(<ButtonGhost>Knopje</ButtonGhost>);

        expect(
            container.firstElementChild?.classList.contains(
                getCssClassNames()[0]
            )
        ).toBeTruthy();
    });
});
