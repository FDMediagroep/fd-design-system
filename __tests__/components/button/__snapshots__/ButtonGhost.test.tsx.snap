// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGhost should handle sizes correctly 1`] = `
<div>
  <button
    class="fdButtonGhost "
  >
    Knopje
  </button>
</div>
`;

exports[`ButtonGhost should handle sizes correctly 2`] = `
<div>
  <button
    class="fdButtonGhost m"
  >
    Knopje
  </button>
</div>
`;
